src/plugins/collapsibleColumns/collapsibleColumns.js

```diff
@@ -408,53 +444,113 @@ class CollapsibleColumns extends BasePlugin {
   /**
    * Collapses/Expands a section.
    *
-   * @param {Object} coords Section coordinates.
+   * @param {Array} coords Array of coords - section coordinates.
    * @param {String} action Action definition ('collapse' or 'expand').
+   * @fires Hooks#beforeColumnCollapse
+   * @fires Hooks#beforeColumnExpand
+   * @fires Hooks#afterColumnCollapse
+   * @fires Hooks#afterColumnExpand
    */
   toggleCollapsibleSection(coords, action) {
-    if (coords.row) {
-      coords.row = parseInt(coords.row, 10);
-    }
-    if (coords.col) {
-      coords.col = parseInt(coords.col, 10);
+    if (!Array.isArray(coords)) {
+      return;
     }
 
+    const currentCollapsedColumns = this.collapsedColumns;
     const hiddenColumns = this.hiddenColumnsPlugin.hiddenColumns;
-    const colspanArray = this.nestedHeadersPlugin.colspanArray;
-    const level = this.nestedHeadersPlugin.rowCoordsToLevel(coords.row);
-    const currentHeaderColspan = colspanArray[level][coords.col].colspan;
-    const childHeaders = this.nestedHeadersPlugin.getChildHeaders(coords.row, coords.col);
-    let nextLevel = level + 1;
-    let childColspanLevel = colspanArray[nextLevel];
-    let firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
-
-    while (firstChildColspan === currentHeaderColspan && nextLevel < this.columnHeaderLevelCount) {
-      nextLevel += 1;
-      childColspanLevel = colspanArray[nextLevel];
-      firstChildColspan = childColspanLevel ? childColspanLevel[childHeaders[0]].colspan || 1 : 1;
-    }
+    const cloneCollapsedSections = new Map(JSON.parse(JSON.stringify(Array.from(this.collapsedSections))));
```

Comment:
```suggestion
    const cloneCollapsedSections = new Map(deepClone(Array.from(this.collapsedSections)));
```
Eariler import the function from the `object.js` helper.